### PR TITLE
udpate access token url on app service

### PR DESCRIPTION
**Pull request by yiliuTo**
Replace parameter from `objectid` to `clientid` of the access token url on App service.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java [hunk 1]
```diff
@@ -135,7 +135,7 @@ private String getAccessTokenOnAppService(String resource, String identity) {
         LOGGER.entering("AuthClient", "getAccessTokenOnAppService", resource);
         LOGGER.info("Getting access token using managed identity based on MSI_SECRET");
         if (identity != null) {
-            LOGGER.log(INFO, "Using managed identity with object ID: {0}", identity);
+            LOGGER.log(INFO, "Using managed identity with client ID: {0}", identity);
```
Comment (chenrujun) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:138]: 1. Update variable name from `identity` to `clientId`.
2. Update java doc.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md [hunk 2]
```diff
@@ -109,7 +109,6 @@ azure:
     tenant-id:           # The Tenant ID for your Azure Key Vault (needed if you are not using managed identity).
     client-id:           # The Client ID that has been setup with access to your Azure Key Vault (needed if you are not using managed identity).
     client-secret:       # The Client Secret that will be used for accessing your Azure Key Vault (needed if you are not using managed identity).
-    # managed-identity:  # The user-assigned managed identity object-id to use.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:112]: We can delete `<!-- embedme...`

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md [hunk 3]
```diff
@@ -141,12 +140,13 @@ Make sure the client-id can access target Key Vault. Here are steps to configure
...
 #### Using a managed identity
 
-If you are using managed identity instead of client-id, add these items in your `application.yml`:
+If you are using managed identity instead of App registrations, add these items in your `application.yml`:
 
 ```yaml
 azure:
   keyvault:
     uri: <the URI of the Azure Key Vault to use>
+    managed-identity: # client-id of the user-assigned managed identity to use. If empty, then system-assigned managed identity will be used.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:149]: Should we comment this line? I mean use `system-assigned managed identity` as default value.

File: sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md [hunk 4]
```diff
@@ -201,11 +200,12 @@ public RestTemplate restTemplateWithTLS() throws Exception {
 
 #### Using a managed identity
 
-If you are using managed identity instead of client-id, add these items in your `application.yml`:
+If you are using managed identity instead of App registration, add these items in your `application.yml`:
 ```yaml
 azure:
   keyvault:
     uri: <the URI of the Azure Key Vault to use>
+    managed-identity:  # client-id of the user-assigned managed identity to use. If empty, then system-assigned managed identity will be used.
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-starter-keyvault-certificates/README.md:208]: Same here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java (hunk 1 above)
Comment (yiliuTo) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:138]: Hi, what does java doc here mean?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java [hunk 5]
```diff
@@ -131,20 +131,20 @@ public String getAccessToken(String resource, String aadAuthenticationUrl,
...
         String result = null;
 
         StringBuilder url = new StringBuilder();
         url.append(System.getenv("MSI_ENDPOINT"))
            .append("?api-version=2017-09-01")
            .append(RESOURCE_FRAGMENT).append(resource);
-        if (identity != null) {
-            url.append("&clientid=").append(identity);
+        if (clientId != null) {
+            url.append("&clientid=").append(clientId);
```
Comment (backwind1233) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:147]: what if move "LOGGER.log(INFO, "Using managed identity with client ID: {0}", clientId);" here

**Review by backwind1233 — COMMENTED** [review]
nice

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java (hunk 5 above)
Comment (yiliuTo) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:147]: Hi, what is your concern of it? I think it's helpful to inform users about the flow of acquiring tokens and also debug.

Comment (chenrujun) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:147]: IMU, @backwind1233 's concern is that `if (clientId != null)` appeared more than one time.

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java (hunk 5 above)
Comment (backwind1233) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/AuthClient.java:147]: @chenrujun  yes 
@yiliuTo 
image

**Pull request merged**